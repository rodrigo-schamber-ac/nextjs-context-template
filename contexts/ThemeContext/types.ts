export enum Theme {
    Portuguese = 'Portuguese',
    English = 'English',
}

export type ThemeContextState = {
    theme: Theme;
    setTheme: (Theme: Theme) => void;
};