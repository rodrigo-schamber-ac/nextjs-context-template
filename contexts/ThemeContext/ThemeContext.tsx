import React, { createContext, useState, FC } from "react";
import { ThemeContextState } from "./types";

const contextDefaultValues: ThemeContextState = {
  mode: "light"
};

export const ThemeContext = createContext<ThemeContextState>(
  contextDefaultValues
);

const ThemeProvider: FC = ({ children }) => {
  const [theme, setTheme] = useState<string>(contextDefaultValues.mode);

  return (
    <ThemeContext.Provider
      value={{
        mode: theme
      }}
    >
      {children}
    </ThemeContext.Provider>
  );
};

export default ThemeProvider;