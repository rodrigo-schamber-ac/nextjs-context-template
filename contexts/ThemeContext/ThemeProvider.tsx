import React, { createContext, useState, FC } from "react";
import { Theme, ThemeContextState } from "./types";

const contextDefaultValues: ThemeContextState = {
  theme: Theme.English,
  setTheme: (theme) => console.warn(`The theme is ${theme}.`)
};

export const ThemeContext = createContext<ThemeContextState>(
  contextDefaultValues
);

export const ThemeConsumer: FC = () => {
  return (
    <ThemeContext.Consumer>
      {(context) => <div>{context.theme}</div>}
    </ThemeContext.Consumer>
  )
};

const ThemeProvider: FC = ({ children }) => {
  const [theme, setTheme] = useState<Theme>(contextDefaultValues.theme);

  return (
    <ThemeContext.Provider
      value={{
        theme, setTheme
      }}
    >
      {children}
    </ThemeContext.Provider>
  );
};

export default ThemeProvider;