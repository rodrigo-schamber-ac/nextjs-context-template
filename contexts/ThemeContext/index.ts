export { ThemeContext, default } from "./ThemeContext";