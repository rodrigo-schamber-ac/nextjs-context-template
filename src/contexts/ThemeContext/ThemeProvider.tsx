import React, { createContext, useState, FC } from 'react';
import { Theme, ThemeContextType } from '../../types/types';

const contextDefaultValues: ThemeContextType = {
  theme: Theme.English,
  setTheme: (theme) => console.warn(`The theme is ${theme}.`)
};

export const ThemeContext =
  createContext<ThemeContextType>(contextDefaultValues);

const ThemeProvider: FC = ({ children }) => {
  const [theme, setTheme] = useState<Theme>(contextDefaultValues.theme);

  return (
    <ThemeContext.Provider
      value={{
        theme,
        setTheme
      }}
    >
      {children}
    </ThemeContext.Provider>
  );
};

export default ThemeProvider;
