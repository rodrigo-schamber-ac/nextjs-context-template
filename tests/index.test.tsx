import React, { FC, ReactElement } from 'react';
import { render, screen } from './test-utils';
import '@testing-library/jest-dom/extend-expect';
import { ThemeConsumer } from '../contexts/ThemeContext';
import ThemeContext from '../contexts/ThemeContext';
import ThemeProvider from '../contexts/ThemeContext/ThemeProvider';

test('ThemeConsumer shows default value', () => {
    render(<ThemeConsumer />)
    expect(screen.getByText(/^English/)).toHaveTextContent(
      'English'
    );
});