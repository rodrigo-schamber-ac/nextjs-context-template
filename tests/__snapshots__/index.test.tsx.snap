// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Context API Take a snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-screen flex justify-center items-center"
      >
        <button
          class="bg-blue-500 text-white px-6 py-2 rounded font-medium mx-3 hover:bg-blue-600 transition duration-200 each-in-out"
          type="button"
        >
          English
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-screen flex justify-center items-center"
    >
      <button
        class="bg-blue-500 text-white px-6 py-2 rounded font-medium mx-3 hover:bg-blue-600 transition duration-200 each-in-out"
        type="button"
      >
        English
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
